### ISSUE TITLE
Grid width stretches off the page

### ISSUE THREAD

**Issue by IeuanWalker** [dotnet/maui#4142]
### Description

A basic grid is stretching off of the page.

The same grid would just fit the width of the page in xamarin.forms.

<table>
<tr>
	<td> Without width request
	<td> With width request
<tr>
	<td> <img src=https://user-images.githubusercontent.com/6544051/149567478-493c340e-68c5-4cdc-8914-7e2d7c1c7f42.png />
	<td><img src=https://user-images.githubusercontent.com/6544051/149567637-e6e0ae09-0dcf-438e-8cd7-859b7156a1e5.png />
</table>

Xaml - 

```xaml 
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage x:Class="FitNote.Pages.SettingsPage"
             xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:vm="clr-namespace:FitNote.ViewModels"
             x:DataType="vm:SettingsViewModel"
             BackgroundColor="{DynamicResource BackgroundColour}"
             NavigationPage.HasNavigationBar="false">
    <ContentPage.Content>
        <ScrollView>
            <VerticalStackLayout Padding="20">
                <Label FontFamily="LASolid"
                       FontSize="40"
                       HorizontalOptions="Start"
                       Text="&#xf104;"
                       TextColor="#191B2E">
                    <Label.GestureRecognizers>
                        <TapGestureRecognizer Tapped="OnBack_Tapped" />
                    </Label.GestureRecognizers>
                </Label>

                <Label Margin="0,20,0,0"
                       FontSize="50"
                       Text="Settings"
                       TextColor="#191B2E" />


                <Label Margin="0,25,0,10"
                       Text="Preferences"
                       TextColor="#191B2E" />
                <!-- Language -->
                <Grid Margin="0,5"
                      ColumnDefinitions="auto,*,auto,auto"
                      HorizontalOptions="Fill"
                      RowDefinitions="auto"
                      RowSpacing="0"
                      WidthRequest="350">
                    <Frame Margin="0"
                           Padding="0"
                           BackgroundColor="#FFF0E7"
                           CornerRadius="15"
                           GridLayout.Column="0"
                           HasShadow="False"
                           HeightRequest="30"
                           WidthRequest="30">
                        <Label FontFamily="LASolid"
                               FontSize="20"
                               HorizontalOptions="Center"
                               Text="&#xf0ac;"
                               TextColor="#E7844E"
                               VerticalOptions="Center" />
                    </Frame>
                    <Label Margin="10,0"
                           GridLayout.Column="1"
                           Text="Language"
                           TextColor="#191B2E"
                           VerticalOptions="Center" />
                    <Label GridLayout.Column="2"
                           HorizontalOptions="End"
                           Text="{Binding Language}"
                           TextColor="#BDBDBF"
                           VerticalOptions="Center" />
                    <Label Margin="5,0"
                           FontFamily="LASolid"
                           FontSize="20"
                           GridLayout.Column="3"
                           HorizontalOptions="End"
                           Text="&#xf105;"
                           TextColor="#191B2E"
                           VerticalOptions="Center" />
                </Grid>
            </VerticalStackLayout>
        </ScrollView>
    </ContentPage.Content>
</ContentPage>
```

### Steps to Reproduce

Repo  -
[repo.zip](https://github.com/dotnet/maui/files/7872510/repo.zip)
 

### Version with bug

Preview 12 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

android 11

**Comment by hartez** [comment]
Should be fixed by #3871

**Comment by IeuanWalker** [comment]
Great thanks, will retest once it's released

**Comment by kristinx0211** [comment]
Verified repro with Android 11. repro sample:
[4142.zip](https://github.com/dotnet/maui/files/8139093/4142.zip)

**Comment by hartez** [comment]
This was fixed by #3871.
